### PR TITLE
Config: Port remaining General settings to new config system.

### PR DESCRIPTION
**Pull request by AdmiralCurtiss**

### CODE REVIEW

File: Source/Core/Core/Config/MainSettings.cpp [hunk 1]
```diff
@@ -163,6 +164,58 @@ const Info<std::string> MAIN_RESOURCEPACK_PATH{{System::Main, "General", "Resour
...
+        {Config::System::Main, "General", fmt::format("ISOPath{}", current_path_idx)}, ""};
+    Config::SetBase(iso_path_info, p);
+    ++current_path_idx;
+  }
+
+  for (size_t i = current_path_idx; i < old_size; ++i)
+  {
+    Config::Info<std::string> iso_path{
+        {Config::System::Main, "General", fmt::format("ISOPath{}", i)}, ""};
+    Config::SetBase(iso_path, "");  // TODO: This actually needs a Config::Erase().
```
Comment (AdmiralCurtiss) [inline Source/Core/Core/Config/MainSettings.cpp:214]: We don't seem to have a way to actually remove a key from the new config system, so I've left this like this for now.

**Comment by AdmiralCurtiss** [comment]
Also note: `IsSettingSaveable` already contained this section key and no specific in- or exclusions, so I didn't need to change anything there.

File: Source/Core/Core/Config/MainSettings.cpp [hunk 2]
```diff
@@ -163,6 +164,58 @@ const Info<std::string> MAIN_RESOURCEPACK_PATH{{System::Main, "General", "Resour
...
+const Info<int> MAIN_ISO_PATH_COUNT{{System::Main, "General", "ISOPaths"}, 0};
+
+std::vector<std::string> GetIsoPaths()
+{
+  size_t count = MathUtil::SaturatingCast<size_t>(Config::Get(Config::MAIN_ISO_PATH_COUNT));
+  std::vector<std::string> paths;
+  paths.reserve(count);
+  for (size_t i = 0; i < count; ++i)
+  {
+    Config::Info<std::string> iso_path_info{
```
Comment (leoetlino) [inline Source/Core/Core/Config/MainSettings.cpp:181]: Can we move this into a helper function (e.g. `MakeISOPathConfigInfo(size_t idx)` or something like that)? This pattern shows up several times.

File: Source/Core/Core/Config/MainSettings.cpp (hunk 1 above)
Comment (leoetlino) [reply Source/Core/Core/Config/MainSettings.cpp:214]: You could call DeleteKey() on the relevant layer but yeah I think this is fine for now.

File: Source/Core/Core/Config/MainSettings.cpp (hunk 2 above)
Comment (AdmiralCurtiss) [reply Source/Core/Core/Config/MainSettings.cpp]: Yup, did that.

**Pull request merged**